app/src/main/java/com/keylesspalace/tusky/components/trending/TrendingPagingAdapter.kt

```diff
@@ -0,0 +1,134 @@
+/* Copyright 2021 Tusky Contributors
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>. */
+
+package com.keylesspalace.tusky.components.trending
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.keylesspalace.tusky.R
+import com.keylesspalace.tusky.adapter.StatusBaseViewHolder
+import com.keylesspalace.tusky.adapter.TagViewHolder
+import com.keylesspalace.tusky.interfaces.LinkListener
+import com.keylesspalace.tusky.util.StatusDisplayOptions
+import com.keylesspalace.tusky.viewdata.TrendingViewData
+
+class TrendingPagingAdapter(
+    private var statusDisplayOptions: StatusDisplayOptions,
+    private val trendingListener: LinkListener,
+) : ListAdapter<TrendingViewData, RecyclerView.ViewHolder>(TrendingDifferCallback) {
+
+    var mediaPreviewEnabled: Boolean
+        get() = statusDisplayOptions.mediaPreviewEnabled
+        set(mediaPreviewEnabled) {
+            statusDisplayOptions = statusDisplayOptions.copy(
+                mediaPreviewEnabled = mediaPreviewEnabled
+            )
+        }
+
+    init {
+        stateRestorationPolicy = StateRestorationPolicy.PREVENT_WHEN_EMPTY
+    }
+
+    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+        return when (viewType) {
```

Comment:
Why the `when` here? These branches do the same thing. Am I missing something?

Reply:
I kept this here as a reminder that other Trending types exist. However, if that isn't clean enough I can remove it. Just let me know.

Reply:
yes please clean it up

Reply:
Cleaned up in the method @connyduck suggests later.